addons/account/models/account_move_line.py

```diff
@@ -448,6 +454,13 @@ class AccountMoveLine(models.Model):
         ),
     ]
 
+    @api.depends('payment_id', 'company_id')
+    def _compute_check_number(self):
+        for record in self:
+            record.check_number = ''
+            if hasattr(record.company_id, 'account_check_printing_date_label') and record.company_id.account_check_printing_date_label:
```

Comment:
Why are you checking `account_check_printing_date_label`?

Reply:
`hasattr(record.company_id, 'account_check_printing_date_label')` if the key presented as it could be not activated.
`record.company_id.account_check_printing_date_label` => No need for it yes, because if it's presented 100% true and if it's not doesn't matter false anyway.

Reply:
>  if it's not doesn't matter false anyway

I don't understand this part.
Why are you not looking for the check number field on the payment instead of a field on the company?

Reply:
I meant that if `account_check_printing_date_label` is not presented here `hasattr(record.company_id, 'account_check_printing_date_label')`, This will mean that the check option is already false (disable in settings)

for your question, I think the check number is something in the company settings that we want to know if the customer have it activated or not and based on that we decide if we want to show the check number column in the reconciliation widget or not.